perfschema/init.go

```diff
@@ -14,344 +14,50 @@
-	{mysql.TypeLonglong, 20, mysql.NotNullFlag | mysql.UnsignedFlag, nil, nil},
-}
-
-var transCurrentCols = []columnInfo{
-	{mysql.TypeLonglong, 20, mysql.NotNullFlag | mysql.UnsignedFlag, nil, nil},
-	{mysql.TypeLonglong, 20, mysql.NotNullFlag | mysql.UnsignedFlag, nil, nil},
-	{mysql.TypeLonglong, 20, mysql.UnsignedFlag, nil, nil},
-	{mysql.TypeVarchar, 128, mysql.NotNullFlag, nil, nil},
-	{mysql.TypeEnum, -1, 0, nil, []string{"ACTIVE", "COMMITTED", "ROLLED BACK"}},
-	{mysql.TypeLonglong, 20, mysql.NotNullFlag | mysql.UnsignedFlag, nil, nil},
-	{mysql.TypeVarchar, 64, 0, nil, nil},
-	{mysql.TypeLong, 11, 0, nil, nil},
-	{mysql.TypeVarchar, 130, 0, nil, nil},
-	{mysql.TypeVarchar, 130, 0, nil, nil},
-	{mysql.TypeVarchar, 64, 0, nil, nil},
-	{mysql.TypeVarchar, 64, 0, nil, nil},
-	{mysql.TypeLonglong, 20, mysql.UnsignedFlag, nil, nil},
-	{mysql.TypeLonglong, 20, mysql.UnsignedFlag, nil, nil},
-	{mysql.TypeLonglong, 20, mysql.UnsignedFlag, nil, nil},
-	{mysql.TypeEnum, -1, 0, nil, []string{"READ ONLY", "READ WRITE"}},
-	{mysql.TypeVarchar, 64, 0, nil, nil},
-	{mysql.TypeEnum, -1, mysql.NotNullFlag, nil, []string{"YES", "NO"}},
-	{mysql.TypeLonglong, 20, mysql.UnsignedFlag, nil, nil},
-	{mysql.TypeLonglong, 20, mysql.UnsignedFlag, nil, nil},
-	{mysql.TypeLonglong, 20, mysql.UnsignedFlag, nil, nil},
-	{mysql.TypeLonglong, 20, mysql.UnsignedFlag, nil, nil},
-	{mysql.TypeLonglong, 20, mysql.UnsignedFlag, nil, nil},
-	{mysql.TypeEnum, -1, 0, nil, []string{"TRANSACTION", "STATEMENT", "STAGE"}},
-}
-
-var stagesCurrentCols = []columnInfo{
-	{mysql.TypeLonglong, 20, mysql.NotNullFlag | mysql.UnsignedFlag, nil, nil},
-	{mysql.TypeLonglong, 20, mysql.NotNullFlag | mysql.UnsignedFlag, nil, nil},
-	{mysql.TypeLonglong, 20, mysql.UnsignedFlag, nil, nil},
-	{mysql.TypeVarchar, 128, mysql.NotNullFlag, nil, nil},
-	{mysql.TypeVarchar, 64, 0, nil, nil},
-	{mysql.TypeLonglong, 20, mysql.UnsignedFlag, nil, nil},
-	{mysql.TypeLonglong, 20, mysql.UnsignedFlag, nil, nil},
-	{mysql.TypeLonglong, 20, mysql.UnsignedFlag, nil, nil},
-	{mysql.TypeLonglong, 20, mysql.UnsignedFlag, nil, nil},
-	{mysql.TypeLonglong, 20, mysql.UnsignedFlag, nil, nil},
-	{mysql.TypeLonglong, 20, mysql.UnsignedFlag, nil, nil},
-	{mysql.TypeEnum, -1, 0, nil, []string{"TRANSACTION", "STATEMENT", "STAGE"}},
-}
-
-func (ps *perfSchema) buildTables() {
-	tbls := make([]*model.TableInfo, 0, len(ps.tables))
-	dbID := autoid.GenLocalSchemaID()
-
-	for name, meta := range ps.tables {
-		tbls = append(tbls, meta)
-		meta.ID = autoid.GenLocalSchemaID()
-		for _, c := range meta.Columns {
-			c.ID = autoid.GenLocalSchemaID()
+// Init register the PERFORMANCE_SCHEMA virtual tables.
+// It should be init(), and the ideal usage should be:
+//
+// import _ "github.com/pingcap/tidb/perfschema"
+//
+// This function depends on plan/core.init(), which initialize the expression.EvalAstExpr function.
+// The initialize order is a problem if init() is used as the function name.
```

Comment:
should we add a `/pkgdep` package in central control  to solve init problem, there are not benefit to use lazy `init()` call in almost situation.

so how about will init pkg that need init ahead.

1. add `github.com/pingcap/tidb/pkgdep` , and add a file `initdep.go`

```
package pkgdep

import (
	
	_ "github.com/pingcap/tidb/planner/core"

	_ "github.com/pingcap/tidb/perfschema"

)
```
2. `in main.go`  just import pkgdep first

```
import (
	"flag"
	"fmt"
	"os"
	"runtime"
	"strconv"
	"strings"
	"sync/atomic"
	"time"

	_ "github.com/pingcap/tidb/pkgdep"

	"github.com/opentracing/opentracing-go"
	"github.com/pingcap/errors"
        .....
```

Reply:
1. Not in this PR
2. I'm not sure whether it works
@lysu